src/Utils.ts

```diff
@@ -1079,24 +1196,20 @@ class DummyInputManagerCore {
             element
         );
 
+        // We will be checking dummy input parents to see if their child list have changed.
+        // So, it is enough to have just one of the inputs observed, because
+        // both dummy inputs always have the same parent.
+        const dummyElement = this._firstDummy.input;
```

Comment:
I suppose that would make it much worse. From the performance perspective it will add much more background payload. Calling a function (`shouldUpdate`) for all instances on all updates is a way heavier than just calling a callback for the subset of instances (created by inline-checking for the presence in a map of changed elements).

Plus `shouldUpdate` would either have to receive the map of changed parents or check the dummy inputs current state all the time, even if they are not in the changed scope.

I try to optimize for performance here as it is constantly running background task. Instead of HTML elements I can track higher level instances. But the actual elements are needed for actual operations (and I wouldn't want to pay the price of uselessly unwrapping the higher level entities all the time in the background task).

Reply:
yeah it's a bit of an overhead I agree, but then why are we trying to encapsulate APIs and constructs when we just go and end up passing raw HTML elements anyway?

I agree about the performance, but until we have more data there it's impossible to make either a claim for or against performance maybe we should add perf tests as a task